functional/ldap_test.go

```diff
@@ -123,3 +125,52 @@ func TestConnectorLDAPHealthy(t *testing.T) {
 		}
 	}
 }
+
+func TestLDAPPool(t *testing.T) {
+	server := ldapServer(t)
+	ldapPool := &connector.LDAPPool{
+		MaxIdleConn: 30,
+		ServerHost:  server.Host,
+		ServerPort:  server.Port,
+		UseTLS:      false,
+		UseSSL:      false,
+	}
+
+	// Excercise pool operations with MaxIdleConn + 10 concurrent goroutines.
+	// The 10 surplus connections should be discarded after use.
+	// Remaining active connections will be verified at the end of the test.
+	numRoutines := ldapPool.MaxIdleConn + 10
+	var wg sync.WaitGroup
+	for i := 0; i < numRoutines; i++ {
+		wg.Add(1)
```

Comment:
The calls are executed serially by the for loop, so no room for races here. But sure, I agree that calling the function 40 times is less efficient than once :-)